packages/toasts/src/ToastsProvider.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { createContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const ToastsContext = createContext();
+
+const ToastsProvider = ({ children }) => {
+  const [toasts, setToasts] = useState([]);
+
+  const addToast = (message) => {
```

Comment:
Right now, `addToast` largely expects a string message, though, technically, a React node could be passed as well (since the `message` gets rendered in the `Toast` children). It might be worth adding a JSDoc docstring above the function to document data types the `message` argument can be.

I also wonder if there's a better function argument name than `message` since I feel `message` kind of implies a string.

Reply:
FWIW the current implementation that we're pulling this from also limits the flexibility of how `Toast` can be used. For example, `Toast` can be optionally customized to include a button/link (see Paragon docs [here](https://paragon-openedx.netlify.app/components/toast/#with-button) and [here](https://paragon-openedx.netlify.app/components/toast/#with-link)) via an `action` prop but the `addToast` examples we have today wouldn't support customizing the `action` prop on a case-by-case basis.

I wonder if there's a possible way to make this functionality more flexible to support that as well while we're in here making changes 🤔